libcontainer/standard_init_linux.go

```diff
@@ -277,7 +278,7 @@ func (l *linuxStandardInit) Init() error {
 
 	if l.dmzExe != nil {
 		l.config.Args[0] = name
-		return system.Fexecve(l.dmzExe.Fd(), l.config.Args, os.Environ())
+		return system.Fexecve(l.dmzExe.Fd(), l.config.Args, l.config.Env)
```

Comment:
Before we merge #4323 , maybe we should include `l.config.Env` when running `StartContainer` hook.

Reply:
This is definitely something to think about. Maybe it makes sense to do it selectively (for those hooks that are run inside the container -- AFAIR not all of them are)

Reply:
Better late than never -- thank you @lifubang, I finally got it now.

This is now fixed and a test case is added.

I still prefer something like #4323 as it's much more straightforward.